hack/prepare-release.sh

```diff
@@ -138,9 +138,15 @@ if [ -z "$no_patching" ]; then
         -i deployment/helm/node-feature-discovery/Chart.yaml
     sed -e s"/pullPolicy:.*/pullPolicy: IfNotPresent/" \
         -e s"!gcr.io/k8s-staging-nfd/node-feature-discovery!registry.k8s.io/nfd/node-feature-discovery!" \
+        -e s"!kubernetes-sigs.github.io/node-feature-discovery/master!kubernetes-sigs.github.io/node-feature-discovery/$docs_version!" \
         -i deployment/helm/node-feature-discovery/values.yaml
     sed -e s"!kubernetes-sigs.github.io/node-feature-discovery/master!kubernetes-sigs.github.io/node-feature-discovery/$docs_version!" \
         -i deployment/helm/node-feature-discovery/README.md
+    sed -e s"!.*metadata\.oci_repo.*!{{- define \"metadata.oci_repo\" -}}oci://registry.k8s.io/nfd/charts/node-feature-discovery!" \
+        -e s"/.*metadata\.oci_tag.*/{{- define \"metadata.oci_tag\" -}}$semver/" \
+        -e s"/.*metadata\.docs_version.*/{{- define \"metadata.docs_version\" -}}$docs_version/" \
+        -i deployment/helm/node-feature-discovery/_metadata.gotmpl
+    make helm-docs
```

Comment:
If the docs isn't in snyc with the values this will fail the release right? Do we want that?

Reply:
I guess this is okay as it is, assuming that now we have this `helm-docs` target call as part of the mandatory CI check. 
But what I was referring to was that, in case if there a mismatch (in the Helm docs) in the HEAD of the repo and we start doing a release, the release process will fail because this script is run when we try to cut a new release and at that point it is too late to ask the merged PR author for the fix. Of course, we can fix it ourselves, and more than that I don't think we will run into that kind of situation since this target is now part of the `scripts/test-infra/verify.sh` too.